### PR TITLE
Simplify Target to use only to cases: BuiltIn and Custom

### PR DESCRIPTION
**Pull request by malbarbo**
In the current form, cross needs to known every target its supports.
This makes things too restrict. For example, its not possible to use a
custom target image for a target that cross does not provides a docker
image (like aarch64-unknown-linux-musl, for example). Cross refuses to
execute saying it does not support the specified target. Besides that,
to add a new docker image the cross code needs to be changed.

This changes simplifies the Target enum to have only two variants:
BuiltIn and Custom. Cross needs to known if a target is a custom target
so it can call xargo, otherwise, all BuitIn targets can be uniformly
handled.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -53,3 +58,19 @@ fn commit_date() -> Result<String, Some> {
...
     }
 }
+
+fn docker_images() -> String {
+    let mut images = String::from("[");
+    let mut dir = PathBuf::from(env::var_os("CARGO_MANIFEST_DIR").unwrap());
+    dir.push("docker");
+    for entry in read_dir(dir).unwrap() {
+        let path = entry.unwrap().path();
+        if path.is_dir() {
```
Comment (Dylan-DPC) [inline build.rs]: we should be using Path(Buf) here instead of depending on string concatenation. Much safer.

Comment (Dylan-DPC) [reply build.rs]: Ah :+1: the path variable misled me.

**Comment by Dylan-DPC** [comment]
Appreciate a second review from @jamesmunns or @rust-embedded/tools

**Comment by Dylan-DPC** [comment]
@malbarbo  Fails on one target
```
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-unknown-dragonfly` target may not be installed
```

**Comment by malbarbo** [comment]
The target x86_64-unknown-dragonfly is not available on rustup, so it also needs xargo. Can we try again?

**Comment by Dylan-DPC** [comment]
Sure. though i feel that having uses_xargo` and `needs_xargo` variables feels a odd and confusing to me. We can refactor it later though.

**Comment by Dylan-DPC** [comment]
bors: r+

**Comment by Dylan-DPC** [comment]
Your latest change broke every platform :smile:

**Comment by malbarbo** [comment]
I've adjusted AvaliableTargets (see the commit message) and remove the needs_xargo flag.

**Comment by Dylan-DPC** [comment]
Dragonfly strikes again. Same issue as we had previously

**Comment by Dylan-DPC** [comment]
bors: try

**Comment by malbarbo** [comment]
The same issue, but a different reason. I tested dragonfly locally. Let's see how is goes.

**Pull request merged**